### ISSUE TITLE
Can you make Breakpoint Modulator update its time mid cycle?

### ISSUE THREAD

**Issue by elanhickler** [RobinSchmidt/RS-MET#208]
When you modulate time scale in breakpoint modulator, the time scale only updates upon next trigger or next loop. Is it possible to make it update immediately or is that the limitation of breakpoint modulator?

**Comment by RobinSchmidt** [comment]
hmm...well...it's not straightforward. possible - maybe - but then perhaps a totally different implementation would be preferable. as it works now, i compute some time-related coefficients at each breakpoint and then they are assumed to be valid until the next breakpoint (i also pre-compute the time-instant of hitting the next breakpoint there). this makes the per-sample computations quite simple (just simple filters and perhaps a few arithmetic operations to create the shapes form the filter-outputs). basically, at each breakpoint hit, i compute a couple of filter coeffs

**Comment by RobinSchmidt** [comment]
for example, the exponential shape is just generated by a simple 1st order filter, the tanh-shape by two 1st order filters - one of them being *unstable* :-O ...which is workable here because i know, when to cut off the response

**Comment by elanhickler** [comment]
Yes! Breakpoint Modulator can be utilized in some interesting ways to create the perfect envelope shape, all we need to do is create some more curve options!! Then I can create the perfect pluck envelope shapes and other envelope shapes I've been interested in. We need the "pluck curve" formula! It's a curve that goes from expo to linear smoothly with a curve option for how smoothly and how drastic of an exponential curve.

**Comment by RobinSchmidt** [comment]
aha! yes - i must think about how to create such a curve. maybe a simple crossfade between exp and linear would do. another interesting curve results from adding two exponential decays with different decay-times (with a crossfade parameter) - this results in an initially fast and later slower decay - a two-stage decay which is typical for piano envelopes, because the initially fast decay is somehow due to interference between the 3 strings per note (as i have read in this challenging book: https://www.amazon.de/Physics-Musical-Instruments-Neville-Fletcher/dp/0387983740)

btw., the bending shape in TriSawModulator is not exp but the bilinear rational formula, but i could add a true exp shape easily. we would then just get two more parameters for the bending-types or something - maybe a combobox, because later we may want to add still more
